lab_1/main.py

```diff
@@ -107,7 +245,15 @@ def load_profile(path_to_file: str) -> dict or None:
     :param path_to_file: a path
     :return: a dictionary with three keys – name, freq, n_words
     """
-    pass
+    if not isinstance(path_to_file, str):
+        return None
+    # load profile from file
+    try:
```

Comment:
can you explain in writing (it's obligatory) how try/except works?

Reply:
We can use block "try" to check the code for errors, in other words, the code inside this block will execute until there is no error in the program. The block "except" will execute if in the block "try" the program encounter some error. In our case, if we try to write path_to_profile that won't exist, we will encounter an error "FileNotFoundError". So we can use try/except to return None in the block "except", writing this type of error. 
Okay, as I understand it, we should check not existing path_to_profile like bad input here, right? I've done this change.

Reply:
that's right. if `if` can be simple, there is no point in using try/except